utils/build-script

```diff
@@ -1040,455 +1513,37 @@ details of the setups of other systems or automated environments.""")
     if args.cmake is not None:
         toolchain.cmake = args.cmake
 
-    if toolchain.cc is None or toolchain.cxx is None:
-        diagnostics.fatal(
-            "can't find clang (please install clang-3.5 or a later version)")
+    # Validate the arguments.
+    BuildScriptInvocation.validate_arguments(toolchain, args)
 
-    if toolchain.cmake is None:
-        diagnostics.fatal("can't find CMake (please install CMake)")
+    # Preprocess the arguments to apply defaults.
+    BuildScriptInvocation.apply_default_arguments(toolchain, args)
```

Comment:
I don't see anything that needs it right now. I have a PR #2961 to reverse the order and infer `--android`.